### PR TITLE
[luci-interpreter] Add support for div op

### PR DESCRIPTION
**Pull request by d-krylov**
Add support for div op in luci-interpreter.

ONE-DCO-1.0-Signed-off-by: Denis Krylov <krylovdenism@gmail.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Div.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+{
+
+Div::Div(const Tensor *input1, const Tensor *input2, Tensor *output, const DivParams &params)
+    : KernelWithParams<DivParams>({input1, input2}, {output}, params)
+{
+}
+
+void Div::configure()
+{
+  LUCI_INTERPRETER_CHECK(input1()->element_type() == input2()->element_type());
```
Comment (ai-moiseev) [inline compiler/luci-interpreter/src/kernels/Div.cpp:35]: There's probably should be one more check against output type:
`LUCI_INTERPRETER_CHECK(input1()->element_type() == output()->element_type())`

File: compiler/luci-interpreter/src/kernels/Div.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  quantizeMultiplier(real_output_multiplier, &output_multiplier, &output_shift);
+
+  int32_t activation_min{};
+  int32_t activation_max{};
+  calculateActivationRangeQuantized(_params.activation, output(), &activation_min, &activation_max);
+
+  tflite::ArithmeticParams params{};
+
+  params.input1_offset = -input1()->zero_point();
+  params.input2_offset = -input2()->zero_point();
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Div.cpp:101]: Please add a comment like in `Add.cpp` regarding the '-'.

File: compiler/luci-interpreter/src/kernels/Div.test.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  Div kernel(&input1_tensor, &input2_tensor, &output_tensor, params);
+  kernel.configure();
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<float>(output_tensor), FloatArrayNear(test_outputs, 0.0001f));
+}
+
+TEST(DivTest, FloatBroadcast)
+{
+  std::initializer_list<int32_t> input1_shape = {1, 3};
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Div.test.cpp:62]: Please use `vector` if there are no specific reasons to use `initializer_list`. In other places too

**Review by s-barannikov — DISMISSED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/Div.test.cpp [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  std::vector<float> input2_data{0.2f, 1.6f, 0.5f};
+  std::vector<float> test_outputs{1.5f, 11.5f, 4.5f, 0.1875f, 1.4375f, 0.5625f, 0.6f, 4.6f, 1.8f};
+
+  Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>(input1_shape, input1_data);
+  Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>(input2_shape, input2_data);
+
+  Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+
+  DivParams params{};
+  params.activation = Activation::RELU;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Div.test.cpp:78]: Please add some negative values to ensure the activation works.

File: compiler/luci-interpreter/src/kernels/Div.test.cpp [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  Tensor input1_tensor =
+      makeInputTensor<DataType::U8>(base_shape, quant_param.first, quant_param.second, input1_data);
+  Tensor input2_tensor =
+      makeInputTensor<DataType::U8>(base_shape, quant_param.first, quant_param.second, input2_data);
+
+  Tensor output_tensor =
+      makeOutputTensor(getElementType<uint8_t>(), quant_param.first, quant_param.second);
+
+  DivParams params{};
+  params.activation = Activation::NONE;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Div.test.cpp:110]: Please use non-default values of the parameters. This will make tests stronger.

**Review by s-barannikov — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/Div.test.cpp [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+{
+namespace
+{
+
+using namespace testing;
+
+float GetTolerance(float min, float max)
+{
+  const float kQuantizedStep = (max - min) / 255.0f;
+  const float kQuantizedTolerance = 2.0f * kQuantizedStep + kQuantizedStep * kQuantizedStep;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Div.test.cpp:33]: Looks like too big tolerance =\

**Review by binarman — APPROVED** [review]
LGTM

**Review by a-zh — APPROVED** [review]
LGTM

**Review by ai-moiseev — APPROVED** [review]
LGTM

**Pull request merged**